clients/src/main/java/org/apache/kafka/common/record/LazyDownConversionRecordsSend.java

```diff
@@ -75,8 +75,7 @@ public long writeTo(GatheringByteChannel channel, long previouslyWritten, int re
                 //      BaseOffset => Int64
                 //      Length => Int32
                 //      ...
-                // TODO: check if there is a better way to encapsulate this logic, perhaps in DefaultRecordBatch
-                log.debug("Constructing fake message batch for topic-partition {" + topicPartition() + "} for remaining length " + remaining);
+                log.debug("Constructing fake message batch for topic-partition {} for remaining length={}", topicPartition(), remaining);
                 int minLength = (Long.SIZE / Byte.SIZE) + (Integer.SIZE / Byte.SIZE);
```

Comment:
I think what you want here is `Records.LOG_OVERHEAD`.

Reply:
What we want to construct here is something that looks like a record batch. Do you think `Records.LOG_OVERHEAD` is appropriate for this case?